spec/plaindate.html

```diff
@@ -406,9 +406,7 @@ <h1>Temporal.PlainDate.prototype.with ( _temporalDateLike_ [ , _options_ ] )</h1
       <emu-alg>
         1. Let _temporalDate_ be the *this* value.
         1. Perform ? RequireInternalSlot(_temporalDate_, [[InitializedTemporalDate]]).
-        1. If Type(_temporalDateLike_) is not Object, then
-          1. Throw a *TypeError* exception.
-        1. Perform ? RejectTemporalLikeObject(_temporalDateLike_).
+        1. If _temporalDateLike_ is not an Object, or ? IsTemporalLikeObject(_temporalDateLike_) is *true*, throw a *TypeError* exception.
```

Comment:
Suggestion: Maybe move the `Object` check into `IsTemporalLikeObject` to avoid checking it before all calls? Similar to how [IsArray](https://tc39.es/ecma262/#sec-isarray) or [IsRegExp](https://tc39.es/ecma262/#sec-isregexp) check for objects in their first step.

Reply:
I thought about that, but it's semantically very different from IsTemporalLikeObject - we'd have to return true from IsTemporalLikeObject if the object was not an Object! I guess we could rename it IsSuitableInputForTemporalWithMethod and reverse the sense of the check.

Reply:
Or, just reverse the sense of the check, if we clarify that IsTemporalLikeObject is not supposed to include actual Temporal objects!

Reply:
Maybe IsPartialTemporalObject